core/src/main/java/org/apache/iceberg/ManifestGroup.java

```diff
@@ -156,6 +156,14 @@ ManifestGroup planWith(ExecutorService newExecutorService) {
    * @return a {@link CloseableIterable} of {@link FileScanTask}
    */
   public CloseableIterable<FileScanTask> planFiles() {
+    return plan((entries, ctx) -> CloseableIterable.transform(entries, entry -> {
+      DataFile dataFile = entry.file().copy(ctx.shouldKeepStats());
+      DeleteFile[] deleteFiles = ctx.deletes().forEntry(entry);
+      return new BaseFileScanTask(dataFile, deleteFiles, ctx.schemaAsString(), ctx.specAsString(), ctx.residuals());
+    }));
```

Comment:
Nit: I kind of prefer a normal function instead of an anonymous one for better readability since we are using the functionInterface here, which needs two steps to figure out the types of input parameters.  Or we just use `BiFunction`.

Reply:
@flyrain, can you elaborate a bit? I am not sure I understood.
Do you mean creating an explicit `CreateFileScanTaskFunction` and using it here?

I did not use `BiFunction` to avoid a bulky definition.

```
public <T extends ScanTask> CloseableIterable<T> plan(BiFunction<CloseableIterable<ManifestEntry<DataFile>>, TaskContext, CloseableIterable<T>> createTasksFunc) {

...
}
```

Reply:
+1 for the new method `createFileScanTasks`. Thanks @aokolnychyi.